src/libcollections/vec_deque.rs

```diff
@@ -2158,6 +2158,18 @@ impl<A: PartialEq> PartialEq for VecDeque<A> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A: Eq> Eq for VecDeque<A> {}
 
+#[stable(feature = "vec-deque-partial-eq-slice", since = "")]
+impl<'a, A: PartialEq> PartialEq<&'a [A]> for VecDeque<A> {
```

Comment:
I can change it to `impl<A: PartialEq> PartialEq<[A]> for VecDeque<A>` and the non-test code compiles.

But if one does:

```rust
let mut d = VecDeque::new();
d.push_back('a');
d.push_back('b');
let v = vec!['a', 'b'];
assert!(d == &v);
```

...which looks perfectly valid to me, this compile error appears:

```
error[E0277]: the trait bound `collections::VecDeque<char>: std::cmp::PartialEq<&collections::vec::Vec<char>>` is not satisfied
   --> src/libcollections/../libcollectionstest/vec_deque.rs:623:13
    |
623 |     assert!(d == &v);
    |             ^^^^^^^ the trait `std::cmp::PartialEq<&collections::vec::Vec<char>>` is not implemented for `collections::VecDeque<char>`
    |
    = help: the following implementations were found:
    = help:   <collections::VecDeque<A> as std::cmp::PartialEq>
    = help:   <collections::VecDeque<A> as std::cmp::PartialEq<[A]>>
```
This does